src/shared.c

```diff
@@ -186,7 +192,7 @@ void move_icon_to_destination(gchar *icon_path, gboolean verbose)
 #endif
         } else {
             g_free(dest_dir);
-            dest_dir = g_build_path("/", g_get_user_data_dir(), "/icons/hicolor/", g_strdup_printf("%ix%i", w, h), "/apps/", NULL);
+            dest_dir = g_build_path("/", xdg_data_home(), "/icons/hicolor/", g_strdup_printf("%ix%i", w, h), "/apps/", NULL);
```

Comment:
xdg_data_home() <- probable memory leak
store the value in a variable before using it, then, free

Reply:
Must've missed this call while fixing the leaks everywhere else. Thanks.